src/components/root_layout.tsx

```diff
@@ -16,20 +19,50 @@ interface RootLayoutProps {
  * @remarks
  * Provides a banner that includes a menu
  */
-const RootLayout: FC<RootLayoutProps> = ({children}) => {
+const RootLayout: FC<RootLayoutProps> = ({ children }) => {
+  const location = useLocation();
+  const [templateName, setTemplateName] = useState("");
+  const [jobName, setJobName] = useState("");
+  const [showBackButtonToTemplate, setShowBackButtonToTemplate] = useState(false);
+
+  useEffect(() => {
```

Comment:
I would separate the idea that the back button should be shown from the idea of where the back button should take you. Setting `showBackButtonToHome` also needs to be set in the useEffect() method.  I notice that `jobName` is not actually used, so we don't need to store it. Here is one possible solution to all this:
```
// templateName set to anything other than the empty string while showBackButton is true indicates that the 
// back button should send the user to the JobsView. templateName set to the empty string while 
// showBackButton is true indicates that the back button should send the user to the Home screen.
const [templateName, setTemplateName] = useState("");
const [showBackButton, setShowBackButton] = useState(false);

useEffect(() => {
    const regexPatternToHome = /^.*?\/app\/[^\/]+$/;
    const regexPatternToTemplate = /^.*?\/app\/([^\/]+)\/[^\/]+$/;
    const toTemplateMatchResult = location.pathname.match(regexPatternToTemplate);
    if (toTemplateMatchResult) {
        setShowBackButton(true);
        const [, capturedTemplateName] = toTemplateMatchResult;
        setTemplateName(capturedTemplateName);
    } else if (regexPatternToHome.test(location.pathname) {
        setShowBackButton(true);
        setTemplateName("");
    } else {
        setShowBackButton(false);
        setTemplateName("");
    }
}, [location.pathname]);
```